packages/java-parser/src/productions/packages-and-modules.js

```diff
@@ -64,16 +64,28 @@ function defineRules($, t) {
     //                 distinguishing between the alternatives due to unbound common prefix.
     // TODO: A post parsing step is required to align with the official specs.
     //       The Identifier "var" is not allowed in all positions and variations of the importDeclaration
-    $.CONSUME(t.Import);
-    $.OPTION(() => {
-      $.CONSUME(t.Static);
-    });
-    $.SUBRULE($.packageOrTypeName);
-    $.OPTION2(() => {
-      $.CONSUME(t.Dot);
-      $.CONSUME(t.Star);
-    });
-    $.CONSUME(t.Semicolon);
+    // Spec Deviation: The spec do not allow empty statement in between imports.
+    //                 However Java compiler consider empty statements valid, we chose
+    //                 to support that case, thus deviate from the spec.
+    $.OR([
+      {
+        ALT: () => {
+          $.CONSUME(t.Import);
+          $.OPTION(() => {
+            $.CONSUME(t.Static);
+          });
+          $.SUBRULE($.packageOrTypeName);
+          $.OPTION2(() => {
+            $.CONSUME(t.Dot);
+            $.CONSUME(t.Star);
+          });
+          $.CONSUME(t.Semicolon);
+        }
+      },
+      {
```

Comment:
The comment should be placed closer to the relevant alternative
and we should include a link to the discussion on this PR.

Reply:
Got it :+1: